### PR TITLE
[CHORE] Move PromiseBelongsTo, PromiseManyArray and ManyArray to model package

### PR DESCRIPTION
**Pull request by Gaurav0**
For #6166 

Also fixes some missing docs

### CODE REVIEW

File: packages/model/addon/index.ts [hunk 1]
```diff
@@ -39,4 +39,4 @@
   @public
  */
 
-export { Model as default, attr, belongsTo, hasMany } from './-private';
```
Comment (runspired) [inline packages/model/addon/index.ts]: We shouldn't re-export private-only things here

File: packages/store/addon/-private/system/promise-proxies.js [hunk 2]
```diff
@@ -85,73 +83,8 @@ export function promiseArray(promise, label) {
...
-
-  reload(options) {
-    assert('You are trying to reload an async belongsTo before it has been created', this.get('content') !== undefined);
-    let { key, store, originatingInternalModel } = this._belongsToState;
-
-    return store.reloadBelongsTo(this, originatingInternalModel, key, options).then(() => this);
-  },
-});
-
 export function proxyToContent(method) {
```
Comment (runspired) [inline packages/store/addon/-private/system/promise-proxies.js]: If we move this too the entanglement will reduce

File: packages/store/index.js [hunk 3]
```diff
@@ -7,6 +7,12 @@ const addonBaseConfig = addonBuildConfigForDataPackage(name);
 module.exports = Object.assign({}, addonBaseConfig, {
   shouldRollupPrivate: true,
   externalDependenciesForPrivateModule() {
-    return ['@ember-data/canary-features', 'ember-inflector', '@ember-data/store/-debug', 'require'];
+    return [
+      '@ember-data/canary-features',
+      'ember-inflector',
+      '@ember-data/store/-debug',
+      'require',
+      '@ember-data/model',
```
Comment (runspired) [inline packages/store/index.js]: if we're doing this, we're breaking encapsulation, so I'm skeptical

File: packages/store/addon/-private/system/model/internal-model.ts [hunk 4]
```diff
@@ -55,14 +64,27 @@ function relationshipStateFor(instance: InternalModel, propertyName: string) {
...
+if (HAS_MODEL_PACKAGE) {
+  _getModelPackage = function() {
+    if (!_found) {
+      let modelPackage = require('@ember-data/model/-private');
+      ({ ManyArray, PromiseBelongsTo, PromiseManyArray } = modelPackage);
+      if (ManyArray && PromiseBelongsTo && PromiseManyArray) {
+        _found = true;
+
+        // Some tests require PromiseManyArray instanceof PromiseArray to be true
+        Object.setPrototypeOf(PromiseManyArray, PromiseArray);
```
Comment (runspired) [inline packages/store/addon/-private/system/model/internal-model.ts]: if we're doing this hack here to fix tests then something unexpected has changed which we should fix

File: packages/model/addon/index.ts (hunk 1 above)
Comment (Gaurav0) [reply packages/model/addon/index.ts]: Fixed

File: packages/store/index.js (hunk 3 above)
Comment (Gaurav0) [reply packages/store/index.js]: Fixed.

**Pull request merged**